radix-engine/src/system/system.rs

```diff
@@ -169,27 +296,279 @@ where
                     &instance_schema.schema,
                     index,
                     SchemaOrigin::Instance,
-                )?;
+                )
+                .map_err(|err| {
+                    RuntimeError::SystemModuleError(
+                        SystemModuleError::BlueprintSchemaValidationError(
+                            BlueprintSchemaValidationError::SchemaValidationError(
+                                err.error_message(&instance_schema.schema),
+                            ),
+                        ),
+                    )
+                })?;
+            }
+        }
+
+        Ok(())
+    }
+
+    // TODO: Move into Package Package
+    pub fn validate_payload_against_blueprint_schema<'s>(
+        &'s mut self,
+        blueprint_id: &BlueprintId,
+        instance_schema: &'s Option<InstanceSchema>,
+        payloads: &[(&Vec<u8>, BlueprintSchemaIdent)],
+    ) -> Result<Vec<(TypePointer, bool)>, RuntimeError> {
+        let blueprint_interface = PackageNativePackage::get_blueprint_default_interface(
+            blueprint_id.package_address.as_node_id(),
+            blueprint_id.blueprint_name.as_str(),
+            self.api,
+        )?;
+
+        let mut schema_pointers = Vec::new();
+
+        // TODO: Move this cache into actor
+        let mut schema_cache = SchemaCache::new();
```

Comment:
Can remove this if have a `get_schema(package_address, hash) -> Option<ScryptoSchema>` system API?

Reply:
How?

Reply:
See https://github.com/radixdlt/radixdlt-scrypto/pull/1114.

By removing `SchemaCache`, we even improved the performance of Radiswap by 2.5%.